.chloggen/elasticsearchexporter_batchsender.yaml

```
@@ -0,0 +1,33 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: elasticsearchexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Improve reliability when used with persistent queue. Remove support for `flush.*`, use `batcher.*` instead.
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [32377]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext: |
+  Move buffering from bulk indexer to batch sender to improve reliability.
```

bike shed, but mostly interested if this is for parity with commit messages or a desire for how release notes/change log reads

I understand rationale of imperative form in commit messages (do this, vs this does this). That said, is it more natural to as a reader of release notes/change log to say the action taking place? 

```suggestion
  Moves buffering from bulk indexer to batch sender to improve reliability.
```

---

I don't mind bike shed. Not a native speaker so it reads identical to me. I use imperative form in git commit messages and therefore the same here.

Would be nice if we can be consistent within this component and even across components. There is no guideline in [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#changelog) on what's preferred. Looking at https://github.com/open-telemetry/opentelemetry-collector-contrib/releases it is ~50/50 at the moment.

---

TL;DR; I suggest we follow Go on commit style, and happy to raise a PR to the CONTRIBUTING file if there are a couple 👍 

Personally, I think release notes are different than commit messages, but I think it helps to consolidate the commit style since you noticed it is 50/50. Release notes, basically I would save to after we agree on commit style ;)

What reads imperative to me ("do this") sounds different if you prefix first. For example, the reason it doesn't sound bad in Go, is because they spell it out. They suggest to think of the prefix "This change modifies Go to _____.", and if that works out, use it as the commit first line (after the prefix of the package affected). You can imagine in this project, we could say "This change modifies the opentelemetry collector *exporter/elasticsearch* to *use exporthelper/batchsender for reliability*. Sounds fine to me!

Here's the doc on Go, and I suggest we use/blame it on them, basically derive and cite this is following go contribution style. https://go.dev/doc/contribute#commit_messages

---

TIL. Yes, I agree on the commit style. It is also close to my usual practice.